python/ray/data/_internal/metadata_exporter.py

```diff
@@ -0,0 +1,218 @@
+"""Metadata exporter API for Ray Data datasets."""
+
+from abc import ABC, abstractmethod
+import logging
+import os
+from dataclasses import dataclass, field
+from typing import Any, List, Optional
+
+import ray
+from ray._private.event.export_event_logger import (
+    EventLogType,
+    get_export_event_logger,
+    check_export_api_enabled,
+)
+
+UNKNOWN = "unknown"
+
+@dataclass
+class SubOperator:
```

Comment:
add a note saying these data classes needs to be in sync with `src/ray/protobuf/export_api/export_data_metadata.proto`?